### DISCUSSION TITLE
Lightning Graph not in stacked columns

### DISCUSSION THREAD

**Discussion by kjames79** [poblabs/weewx-belchertown#894] [Q&A]
I used the Charts Document to create my lightning graph and i want it to be stacked columns the way it is displayed in that document but I am not getting that.

Here is my configuration:

    [[chart8]]
        title = Lightning
        [[[lightning_strike_count]]]
            name = Lightning
            stacking = normal
        [[[lightning_distance]]]
            name = Lightning_Distance
            stacking = normal

what I have: <a href="https://imgur.com/ZcAPtiW"><img src="https://i.imgur.com/ZcAPtiW.jpg" title="source: imgur.com" /></a>

what I want: :<a href="https://imgur.com/bkP48Zl"><img src="https://i.imgur.com/bkP48Zl.png" title="source: imgur.com" /></a>

**Comment by poblabs** [comment]
I think you're missing the `type` option to specify the type you want. Here's what I have on belchertownweather.com - adjust it for your needs

```
    [[lightninggroupby]]
        title = Lightning Strikes This Year
        time_length = year
        xAxis_groupby = month
        xAxis_categories = 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'

        [[[lightning_distance]]]
            name = Average Distance (mi)
            type = spline
            aggregate_interval = 300 # 5 minutes
            aggregate_type = avg
            yAxis = 1
            yAxisLabel = Average Distance (mile)
            zIndex = 1
            [[[[marker]]]]
                enabled = true
                radius = 4
        [[[lightning_strike_count]]]
            name = Lightning Count
            type = column
            yAxisLabel = Lightning Count
```

![image](https://github.com/poblabs/weewx-belchertown/assets/3484775/3b81f2bb-f466-4b37-9a7c-dfcf21cf3475)

**Comment by kjames79** [comment]
ok so that got me almost exactly where I need to be. Thank you! I'd just like to get rid of the connecting line for average distance. If it's not possible that's fine. This is already much better

**Comment by kjames79** [comment]
ok so I think I'm getting pretty close here. I have the stacking the way I want it and I have closest strike instead of average, just my preference. Where I'm having trouble is I have total strikes showing up on days when there wasn't any lightning. In this screenshot there is only one day where there was lightning. that's the only day I want a stat to show up. The other problem I have that I can't figure out how to correct is the way time is displayed. If you look in this screenshot you'll see that for Rain I have a date with a month and a noon marker. I'm not sure why that is how all my other graphs show but I like it better than the way the Lightning is shown. I'm not actually sure what the lightning is showing. it looks like the 3rd but the rain took place on the 4th. I'm confused by this. Here's what I have for both

    [[chart4]]
        title = Rain
        [[[rainRate]]]
            yAxis = 1
        [[[rainTotal]]]
#            name = Rain Total

    [[lightninggroupby]]
        title = Lightning
        time_length = 604800 # Last 7 days
        xAxis_groupby = day


        [[[lightning_distance]]]
            name = Closest Strike
            type = column
            stacking = normal
            aggregate_interval = 300 # 5 minutes
            aggregate_type = min
            yAxis = 1
            yAxisLabel = Closest Distance (mile)
            zIndex = 1
            
            [[[[marker]]]]
                enabled = true
                radius = 4

        [[[lightning_strike_count]]]
            name = Lightning Count
            type = column
            stacking = normal
            yAxisLabel = Lightning Count


<a href="https://imgur.com/46lxluF"><img src="https://i.imgur.com/46lxluF.jpg" title="source: imgur.com" /></a>
